src/main/java/emt/item/armor/boots/ItemElectricBootsTraveller.java

```diff
@@ -117,19 +120,29 @@ public int getVisDiscount(ItemStack stack, EntityPlayer player, Aspect aspect) {
 
     @Override
     public void onArmorTick(World world, EntityPlayer player, ItemStack itemStack) {
-        if ((!player.capabilities.isFlying) && (player.moveForward > 0.0F)) {
-            if ((player.worldObj.isRemote) && (!player.isSneaking())) {
-                if (!Thaumcraft.instance.entityEventHandler.prevStep.containsKey(
-                        Integer.valueOf(player.getEntityId()))) {
-                    Thaumcraft.instance.entityEventHandler.prevStep.put(
-                            Integer.valueOf(player.getEntityId()), Float.valueOf(player.stepHeight));
+        if (!player.capabilities.isFlying && player.moveForward > 0.0F) {
+            if (player.worldObj.isRemote && !player.isSneaking()) {
+                if (!Thaumcraft.instance.entityEventHandler.prevStep.containsKey(player.getEntityId())) {
+                    Thaumcraft.instance.entityEventHandler.prevStep.put(player.getEntityId(), player.stepHeight);
                 }
                 player.stepHeight = 1.0F;
             }
 
-            if (((player.onGround || player.capabilities.isFlying) && player.moveForward > 0F)) {
-                player.moveFlying(0F, 1F, speedBonus);
-                player.jumpMovementFactor = jumpBonus;
+            if (player.onGround) {
+                float bonus = speedBonus;
+                if (player.isInWater()) {
+                    bonus /= 4.0F;
+                }
+
+                player.moveFlying(0.0F, 1.0F, bonus);
+            } else if (Hover.getHover(player.getEntityId())) {
+                // Base ItemBootsTraveller jumpBonus equals to jumpBonus of Electric Boots,
+                // so any other boots factor can be calculated via proportion method
+                player.jumpMovementFactor =
+                        0.03F / ((ItemElectricBootsTraveller) EMTItems.electricBootsTraveller).jumpBonus * jumpBonus;
```

Comment:
Is it intended that quantum boots will have *smaller* jump factor than nano boots?